frontend/src/SettingsDashboard.jsx

```diff
@@ -75,27 +90,54 @@ class NewSettingsModalInner extends React.Component {
         </Modal>;
     }
 }
-const NewSettingsModal = connect(null, mapDispatchToProps)(NewSettingsModalInner);
+const SettingsModal = connect(null, mapDispatchToProps)(SettingsModalInner);
 
-const SettingsTable = ({ name, settings, settingsType, onEditCell }) => {
+
+const SettingsTableInner = ({ name, settings, settingsType, onEditCell, deleteSetting }) => {
     const [modalOpen, setModalOpen] = React.useState(false);
```

Comment:
For this kind of situation I prefer to use a state that takes values like this `modalOpen = null | "Update" | "New"`
It makes the handling of the modal a little more straightforward, especially when we can expect having different modals or configurations of them. This is not the case here though so good to keep like this.

Reply:
I much prefer that - we can have one piece of state control both the open/close status as well as the "type" of modal.